File: app/src/main/java/com/polok/eubmanagement/base/BaseApp.java
package com.polok.eubmanagement.base;

import android.app.Application;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.polok.eubmanagement.util.SharedPref;

public class BaseApp extends Application {
    private static DatabaseReference databaseReference;

    @Override
    public void onCreate() {
        super.onCreate();
        databaseReference = FirebaseDatabase.getInstance().getReference();
        SharedPref.init(getApplicationContext());
    }

    public static DatabaseReference getFirebaseDataRef() {
        return databaseReference;
    }
}

File: app/src/main/java/com/polok/eubmanagement/base/BaseApp.kt
package com.polok.eubmanagement.base

import android.app.Application
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.polok.eubmanagement.util.SharedPref

class BaseApp : Application() {
    override fun onCreate() {
        super.onCreate()
        firebaseDataRef = FirebaseDatabase.getInstance().getReference()
        SharedPref.init(applicationContext)
    }

    companion object {
        var firebaseDataRef: DatabaseReference? = null
            private set
    }
}
